src/pages/mentor-home/MentorHome.js

```diff
@@ -1,14 +1,18 @@
 import { useEffect, useContext } from 'react'
+import { useSelector } from 'react-redux'
 import BecomeATutor from '~/containers/mentor-home-page/become-a-tutor/BecomeATutor'
 import { ModalContext } from '~/context/modal-context'
 
 const MentorHome = () => {
   const { setModal, setFullScreen } = useContext(ModalContext)
+  const { isFirstLogin } = useSelector((state) => state.appMain)
 
   useEffect(() => {
-    setFullScreen(true)
-    setModal(<BecomeATutor />)
-  }, [setModal, setFullScreen])
+    if (isFirstLogin) {
+      setFullScreen(true)
```

Comment:
How fullscreen looks like ?

Reply:
I don't think we need a "useEffect" hook here.

Reply:
The hook was used before in code, to show the modal on render of this component. Now I need to check for this condition on render of this component. How can I do that without useEffect? I just didn’t know there were other ways

Reply:
Place this if in the component without hooks at all :)

Reply:
I tried adding it, but there was a “Maximum depth exceeded error”, which I assume was due to multiple rerendering, so I guess I can’t really avoid using useEffect here

Reply:
Let's discuss it on our sync-up.